File: games/dice.py
# games/dice.py
import random

class DiceEvenOdd:
    def roll_dice(self):
        return random.randint(1, 6), random.randint(1, 6)

    def play_round(self, player_choice, bet, balance):
        d1, d2 = self.roll_dice()
        total = d1 + d2
        parity = "even" if total % 2 == 0 else "odd"

        if player_choice == parity:
            balance += bet
            result = "win"
            message = f"Dice: {d1} + {d2} = {total} ({parity}). You win {bet}!"
        else:
            balance -= bet
            result = "loss"
            message = f"Dice: {d1} + {d2} = {total} ({parity}). You lose {bet}."

        return {
            "game": "dice",
            "player_choice": player_choice,
            "computer_choice": parity,
            "extra": {"d1": d1, "d2": d2, "total": total},
            "result": result,
            "balance": balance,
            "message": message
        }

File: static/dice.js
let diceBalance = 100, diceWins = 0, diceLosses = 0;
let diceActive = true;

document.addEventListener("DOMContentLoaded", () => {
  const balanceSpan = document.getElementById("diceBalance");
  const betInput = document.getElementById("diceBet");
  const statusP = document.getElementById("diceStatus");
  const winsSpan = document.getElementById("diceWins");
  const lossesSpan = document.getElementById("diceLosses");
  const timeSpan = document.getElementById("diceTime");
  const display = document.getElementById("diceDisplay");
  const exitBtn = document.getElementById("diceExit");
  const optionButtons = document.querySelectorAll(".option-btn");

  balanceSpan.textContent = diceBalance;

  exitBtn.addEventListener("click", () => {
    alert(`You exited Dice with ${diceBalance} chips.`);
    window.location.href = "/";
  });

  optionButtons.forEach(btn => {
    btn.addEventListener("click", () => {
      if (!diceActive) {
        alert("Session ended. Refresh to play again.");
        return;
      }
      if (diceBalance <= 0) {
        alert("No balance left.");
        diceActive = false;
        return;
      }

      const choice = btn.getAttribute("data-choice");  // "even" or "odd"
      const bet = parseInt(betInput.value, 10) || 0;
      if (bet <= 0 || bet > diceBalance) {
        alert("Enter valid bet ≤ balance.");
        return;
      }

      fetch("/play", {
        method: "POST",
        headers: { "Content-Type": "application/json" },
        body: JSON.stringify({
          game: "dice",
          choice: choice,
          bet: bet,
          balance: diceBalance,
          wins: diceWins,
          losses: diceLosses,
          ties: 0
        })
      })
      .then(res => res.json())
      .then(data => {
        diceBalance = data.balance;
        if (data.result === "win") diceWins++; else diceLosses++;

        balanceSpan.textContent = diceBalance;

        const extra = data.extra || {};
        const d1 = extra.d1 ?? "–";
        const d2 = extra.d2 ?? "–";
        const total = extra.total ?? "–";
        const parity = data.computer_choice ? data.computer_choice.toUpperCase() : "";
        display.textContent = `${d1} + ${d2} = ${total} (${parity})`;

        winsSpan.textContent = diceWins;
        lossesSpan.textContent = diceLosses;
        statusP.textContent = data.message;
        timeSpan.textContent = data.time;

        if (diceBalance <= 0) {
          alert("You are out of chips in Dice.");
          diceActive = false;
        }
      });
    });
  });
});
